bidscoin/heuristics/bidsmap_dccn.yaml

```diff
@@ -40,6 +40,11 @@ Options:
       meta: [.json, .tsv, .tsv.gz]  # The file extensions of the equally named metadata sourcefiles that are copied over to the BIDS sidecar files
       multiraid: 2                  # The mapVBVD argument for selecting the multiraid Twix file to load (default = 2, i.e. 2nd file)
 #    phys2bidscoin:                  # The settings for the phys2bidscoin plugin
+    dcm2niix4petbids:
+      command: dcm2niix4pet
+      args:
+      anon:
+      meta: [.json, .tsv]
```

Comment:
I ended up adding the plugin here as it seemed like as good a place as any. I am curious as to whether or not I could adopt the more modular approach that is shown in the example plugin in the read the docs. See

```# The default bids-mappings that are added when installing the plugin
BIDSMAP = {'DemoFormat':{
    'subject': '<<filepath:/sub-(.*?)/>>',     # This filesystem property extracts the subject label from the source directory. NB: Any property or attribute can be used as subject-label, e.g. <PatientID>
    'session': '<<filepath:/ses-(.*?)/>>',     # This filesystem property extracts the session label from the source directory. NB: Any property or attribute can be used as session-label, e.g. <StudyID>

    'func': [                   # ----------------------- All functional runs --------------------
        {'provenance': '',      # The fullpath name of the source file from which the attributes and properties are read. Serves also as a look-up key to find a run in the bidsmap
         'properties':          # The matching (regexp) criteria go in here
            {'filepath': '',    # File folder, e.g. ".*Parkinson.*" or ".*(phantom|bottle).*"
             'filename': '',    # File name, e.g. ".*fmap.*" or ".*(fmap|field.?map|B0.?map).*"
             'filesize': '',    # File size, e.g. "2[4-6]\d MB" for matching files between 240-269 MB
             'nrfiles': ''},    # Number of files in the folder that match the above criteria, e.g. "5/d/d" for matching a number between 500-599
         'attributes':          # The matching (regexp) criteria go in here
            {'ch_num': '.*',
             'filetype': '.*',
             'freq': '.*',
             'ch_name': '.*',
             'units': '.*',
             'trigger_idx': '.*'},
         'bids':
            {'task': '',
             'acq': '',
             'ce': '',
             'dir': '',
             'rec': '',
             'run': '<<1>>',    # This will be updated during bidscoiner runtime (as it depends on the already existing files)
             'recording': '',
             'suffix': 'physio'},
         'meta':                # This is an optional entry for meta-data dictionary that are appended to the json sidecar files
            {'TriggerChannel': '<<trigger_idx>>',
             'ExpectedTimepoints': '<<num_timepoints_found>>',
             'ChannelNames': '<<ch_name>>',
             'Threshold': '<<thr>>',
             'TimeOffset': '<<time_offset>>'}}],

    [...]
```

But I knew I could get things working by adding them to the main template in `bidscoin/heuristics/bidsmap_dccn.yaml`

Reply:
True, I did a first implementation of that some time ago, but I never really tested it very well. The template bidsmap is getting a bit big (with every new plugin), so perhaps it's time to refactor this

Reply:
The advantage of the bidsmap is that every site can have it's own, and it doesn't require modifying the plugin

Reply:
Hmm, maybe we should bump all of the plugins out of `bidsmap_dccn.yaml` into each corresponding `plugin.py` file in a separate PR. Then it's not quite so ad hoc.

Reply:
Yes, a separate (dedicated) bidsmap per plugin. It's a bit more complicated to handle but it could be an improvement to the maintenance and modularity on the long run

Reply:
My original thought / idea was that multiple plugin could work using the same mappings (e.g. both dcm2niix2bids and dcm2niix4pet would work with the same DICOM section of the bidsmap), but I guess that doesn't fly very well

Reply:
Separate the logic from the code, it's just that all the logic went into one bidsmap, so that each study could have one bidsmap file with all the info in it to produce the BIDS output

Reply:
Like I said, I think my wanting to use that feature could wait until later.